Fix edge directionality in graph layout

fbarl opened the pull request:
Fixes #2267.

![layout-fixed](https://cloud.githubusercontent.com/assets/1216874/23211651/69841f72-f903-11e6-86d4-1d3a2327d282.png)

The problem was that we were not making a difference between *source* and *target* nodes when constructing edges for the layout engine (for the purpose of not rendering the same edge twice).

~~**Note:** The bidirectional edges are now represented as two directed edges, both in the layout engine and when rendering. So they are actually rendered as two overlapping edges, although that's currently not visible. I intentionally left it like that so that we could make them non-overlapping if we want or have an easier time drawing the arrows later on.~~

rade commented:
> The bidirectional edges are now represented as two directed edges, both in the layout engine and when rendering. So they are actually rendered as two overlapping edges, although that's currently not visible. I intentionally left it like that so that we could make them non-overlapping if we want or have an easier time drawing the arrows later on.

Hmm. Not sure about that. I would expect bidirectional edges to be rendered as one edge with two arrowheads.

If it's relatively easy to have one edge for bidirectional edges, let's do that. If not, then what you have is ok.

rade commented:
btw, was this always broken, or is it a regression?

fbarl commented:
@rade Alright, I modified the code so that it doesn't render both edges, but I'm still sending them both to the layout engine so that the rank assignment would be more accurate. I cleaned up that code and added some comments and unit tests.

I think I broke edge directions in one of my optimizations stories, so I don't think it was there for too long. However, I think we were always ignoring one direction in case of bidirectional edges, which is something this story also fixes. There is a chance this fix will improve stability and accuracy of some rendered graphs slightly.

rade commented:
> I think I broke edge directions in one of my optimizations stories,

Did that breakage make it 1.2? If so, we *may* want to release a 1.2.1 with a fix. cc @2opremio

jpellizzari commented on client/app/scripts/utils/__tests__/layouter-utils-test.js:
```diff
@@ -0,0 +1,41 @@
+import { fromJS } from 'immutable';
+
+describe('LayouterUtils', () => {
+  const LayouterUtils = require('../layouter-utils');
```
`import` syntax would keep this more consistent

fbarl replied:
Agree

fbarl commented:
@rade Fortunately the breakage didn't make it into any of the releases as it was introduced on *master* only 2 days ago when merging #2221 (line 19 in [nodes-chart-layout.js](https://github.com/weaveworks/scope/pull/2221/files#diff-bb937cf1c7e428785c32dca6802273bc)).

jpellizzari commented:
LGTM. Nice to have tests for this!
